examples/placegoose/src/lib/validation.ts

```diff
@@ -0,0 +1,55 @@
+import type { Context } from "hono";
+import { HTTPException } from "hono/http-exception";
+
+/**
+ * @returns Validation fn for Hono body validator, responsible
+ * for processing payload errors
+ */
+export function makeBodyValidator<T extends Record<string, unknown>>(
+  parse: (data: unknown) => T,
+) {
+  return (body: unknown) => {
+    try {
+      // Return value must be consistent with shape of "body"
+      return parse(body);
+    } catch (error) {
+      console.error(error);
+      throw new HTTPException(400, {
+        message: "Invalid Payload",
+        cause: error,
+      });
+    }
+  };
+}
+
+/**
+ * Verify that query or param value is valid id, and
+ * coerce to number. Throws if id is missing or invalid
+ * @returns Valid number id
+ */
+export function validateId(value: string | string[]) {
+  if (typeof value !== "string" || !/^[1-9]\d*$/.test(value)) {
+    throw new HTTPException(400, {
+      message: "ID values must be positive integers",
+    });
+  }
+
+  return Number(value);
```

Comment:
nit: `Number("")` returns `0` in javascript, so i typically avoid it when coercing numbers, but you **do** handle this case above with your regex

`+value` would technically be slower, but at least would return a `Nan` for emptystrings

otherwise, `Number.parseInt` is probably the safest option here